sklearn/naive_bayes.py

```diff
@@ -379,8 +414,11 @@ def _partial_fit(self, X, y, classes=None, _refit=False,
             self.sigma_[i, :] = new_sigma
             self.class_count_[i] += N_i
 
+        class_prior = self.class_prior
+
         self.sigma_[:, :] += epsilon
-        self.class_prior_[:] = self.class_count_ / np.sum(self.class_count_)
+        self._update_class_prior(class_prior=class_prior)
```

Comment:
Yes, I think we should update the prior only when

    if self.class_prior is None and self.fit_prior

and `self.class_prior_` to be constant otherwise.

I would just do it at fit time rather than at the constructor. (One reason is while cloning the estimator we want only the hyperparameters to be copied and not those parameters that are explicitly set)